File: solutions/golang/128/solution.go
func longestConsecutive(nums []int) int {
	count := make(map[int]bool)
	for _, num := range nums {
		count[num] = true
	}

	longest := 0
	for len(count) > 0 {
		var start int
		for num := range count {
			_, ok := count[num-1]
			if !ok {
				start = num
				break
			}
		}
		delete(count, start)
		lenSoFar := 1
		_, ok := count[start+1]
		for ; ok == true; _, ok = count[start+1] {
			delete(count, start+1)
			lenSoFar++
			start++
		}
		longest = max(longest, lenSoFar)
	}

	return longest
}

File: solutions/cpp/128/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> count;
        for (int num : nums) {
            count.insert(num);
        }
        int longest = 0;
        while (count.size()) {
            int start;
            for (int num : count) {
                if (count.find(num - 1) == count.end()) {
                    start = num;
                    break;
                }
            }
            int lenSoFar = 1;
            count.erase(start);
            while (count.find(start + 1) != count.end()) {
                count.erase(start + 1);
                lenSoFar++;
                start++;
            }
            longest = max(longest, lenSoFar);
        }
        return longest;
    }
};